db/migrate/20180824155031_create_recovery_codes.rb

```diff
@@ -0,0 +1,10 @@
+class CreateRecoveryCodes < ActiveRecord::Migration[5.1]
+  def change
+    create_table :recovery_codes do |t|
+      t.integer :user_id, null: false
+      t.text :code, null: false
+      t.integer :used, default: 0, null: false
```

Comment:
This feels more like a boolean than an integer, but if we track when it was used, then we don't really need the boolean.